modules/RTC/MungeLocalSdp.js

```diff
@@ -0,0 +1,295 @@
+                            : this.pc.simulcast.ssrcCache;
+                    if (!requiredSSRCs.length) {
+                        logger.error(
+                            "No SSRCs stored for: "
+                                + videoTrack + " in " + this.pc);
+                        return;
+                    }
+                    if (!transformer.getSSRCCount()) {
+                        logger.error(
+                            "No video SSRCs found "
+                                + "(should be at least the recv-only one");
+                        return;
+                    }
+
+                    modified = true;
+
+                    // We need to fake sendrecv
+                    transformer.mediaDirection = "sendrecv";
+
+                    // Check if the recvonly has MSID
+                    const primarySSRC = requiredSSRCs[0];
+                    // FIXME the cname could come from the stream, but may
+                    // turn out to be too complex. It is fine to come up
+                    // with any value, as long as we only care about
+                    // the actual SSRC values when deciding whether or not
+                    // an update should be sent
+                    const primaryCname = "injected-" + primarySSRC;
+
+                    requiredSSRCs.forEach((ssrcNum) => {
+                        // Remove old attributes
+                        transformer.removeSSRC(ssrcNum);
+                        // Inject
+                        logger.debug(
+                            "Injecting video SSRC: " + ssrcNum
+                                + " for " + videoTrack);
+                        transformer.addSSRCAttribute({
+                            id: ssrcNum,
+                            attribute: 'cname',
+                            value: primaryCname
+                        });
+                        transformer.addSSRCAttribute({
+                            id: ssrcNum,
+                            attribute: 'msid',
+                            value: videoTrack.storedMSID
+                        });
+                    });
+                    if (requiredSSRCs.length > 1) {
+                        const group = {
+                            ssrcs: requiredSSRCs.join(" "),
+                            semantics: "SIM"
+                        };
+                        if (!transformer.findGroup(
+                                group.semantics, group.ssrcs)) {
+                            // Inject the group
+                            logger.debug(
+                                "Injecting SIM group for " + videoTrack, group);
+                            transformer.addSSRCGroup(group);
+                        }
+                    }
+                    // Insert RTX
+                    // FIXME in P2P RTX is used by Chrome regardless of this
+                    // option status
+                    if (!this.pc.options.disableRtx) {
```

Comment:
From what I could understand about RtxModifier it's responsibility is to make sure that RTX and SSRCs are present in the SDP and that they remain the same, after unmuting video track. The purpose of this class is to fake localDescription to prevent from sending source-add/source-remove and get rid of "modified SSRC info" state. It is also faking the SDP for detached tracks. I'm not sure what you mean exactly by leveraging the logic. Is it about exposing some methods and using them here ?